cmd/compgen/cmd/receiver/templates/config.go.tmpl

```diff
@@ -14,16 +14,16 @@ import (
 const configKey = "sample"
 
 var (
-	ErrMustString    = errors.New("sample.configuration must be a string")
-	ErrMustNotNil    = errors.New("sample interface must not be nil")
-	ErrSampleConfig  = errors.New("sample config data is required")
-	ErrMustLowercase = errors.New("sample config data must be lowercase")
+	ErrMustString    = "'sample' expected type 'string'"
+	ErrMustNotNil    = "sample interface must not be nil"
+	ErrSampleConfig  = "sample config data is required"
+	ErrMustLowercase = "sample config data must be lowercase"
```

Comment:
The switch to error strings was because [this test](https://github.com/liatrio/liatrio-otel-collector/blob/7839d171c2e7f4d9fe58a8733f6021f8f8e9fcf7/cmd/compgen/cmd/receiver/templates/config_test.go.tmpl#L69-L76) was returning an error from [here](https://github.com/liatrio/liatrio-otel-collector/blob/7839d171c2e7f4d9fe58a8733f6021f8f8e9fcf7/cmd/compgen/cmd/receiver/templates/config.go.tmpl#L36). confmap.Unmarshall was bubbling the error up from mapstructure.Decode which did not have any public error variables for comparison.